python/orca/src/bigdl/orca/ray/raycontext.py

```diff
@@ -36,18 +36,19 @@ def __init__(self,
             from bigdl.orca.ray import RayOnSparkContext
             self._ray_on_spark_context = RayOnSparkContext(**kwargs)
             self.is_local = self._ray_on_spark_context.is_local
+            self.num_ray_nodes = self._ray_on_spark_context.num_ray_nodes
```

Comment:
You may not necessarily have `_ray_on_spark_context` in this case?

Instead of `num_ray_nodes` and `ray_node_cpu_cores`, make it more general: `OrcaContext.node_num`, `OrcaContext.core_num`

Reply:
> You may not necessarily have `_ray_on_spark_context` in this case?
> 
> Instead of `num_ray_nodes` and `ray_node_cpu_cores`, make it more general: `OrcaContext.node_num`, `OrcaContext.core_num`

1. When the `runtime` is spark, init_orca_context will also create a rayonspark context object.

3. In this case, pytorch_ray_estimator on ray backend will calculate using these 2 parameters here https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/learn/pytorch/pytorch_ray_estimator.py#L151. Maybe we could also save the logical to get `ray_ctx.num_ray_nodes` in this PR ?

Reply:
> > You may not necessarily have `_ray_on_spark_context` in this case?
> > Instead of `num_ray_nodes` and `ray_node_cpu_cores`, make it more general: `OrcaContext.node_num`, `OrcaContext.core_num`
> 
> 1. When the `runtime` is spark, init_orca_context will also create a rayonspark context object.
> 2. In this case, pytorch_ray_estimator on ray backend will calculate using these 2 parameters here [https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/learn/pytorch/pytorch_ray_estimator.py#L151](https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/learn/pytorch/pytorch_ray_estimator.py?rgh-link-date=2022-04-19T01%3A58%3A41Z#L151). Maybe we could also save the logical to get `ray_ctx.num_ray_nodes` in this PR ?

We just need to know the cluster resources (node#, core#) to correctly compute the info, yes?

Reply:
> > > You may not necessarily have `_ray_on_spark_context` in this case?
> > > Instead of `num_ray_nodes` and `ray_node_cpu_cores`, make it more general: `OrcaContext.node_num`, `OrcaContext.core_num`
> > 
> > 
> > 
> > 1. When the `runtime` is spark, init_orca_context will also create a rayonspark context object.
> > 2. In this case, pytorch_ray_estimator on ray backend will calculate using these 2 parameters here [https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/learn/pytorch/pytorch_ray_estimator.py#L151](https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/learn/pytorch/pytorch_ray_estimator.py?rgh-link-date=2022-04-19T01%3A58%3A41Z#L151). Maybe we could also save the logical to get `ray_ctx.num_ray_nodes` in this PR ?
> 
> We just need to know the cluster resources (node#, core#) to correctly compute the info, yes?

Yes, that's right.